core/arch/arm/kernel/mutex.c

```diff
@@ -60,6 +66,29 @@ static void __mutex_lock(struct mutex *m, const char *fname, int lineno)
 	}
 }
 
+static void __mutex_lock_recursive(struct recursive_mutex *m, const char *fname,
+				   int lineno)
+{
+	assert_have_no_spinlock();
+	assert(thread_get_id_may_fail() != THREAD_ID_INVALID);
+	assert(thread_is_in_normal_mode());
+
+	/* Thread ID has to fit in a short int */
+	COMPILE_TIME_ASSERT(CFG_NUM_THREADS <= SHRT_MAX);
+
+	if (atomic_load_short(&m->owner) == thread_get_id()) {
```

Comment:
Perhaps we could save the return value of `thread_get_id()` in a local variable since we're calling this function again below and we're supposed to get the same result.